src/coprocessor/endpoint.rs

```diff
@@ -118,104 +109,69 @@ impl Host {
         // Otherwize collect into thread local contexts.
         let metrics = &mut self.basic_local_metrics;
 
-        if let Err(e) = t.check_outdated() {
+        if let Err(e) = t.req_ctx.check_if_outdated() {
             let resp = err_resp(e, metrics);
             t.on_resp.respond(resp);
             return;
         }
 
-        let (mut req, cop_req, req_ctx, on_resp) = (t.req, t.cop_req, t.ctx, t.on_resp);
-        let mut tracker = t.tracker;
-        let ranges = req.take_ranges().into_vec();
+        let (mut tracker, req_ctx, req_handler_builder, on_resp) =
+            (t.tracker, t.req_ctx, t.req_handler_builder, t.on_resp);
 
-        let priority = readpool::Priority::from(req.get_context().get_priority());
+        let priority = readpool::Priority::from(req_ctx.context.get_priority());
         let pool = self.pool.get_pool_by_priority(priority);
         let ctxd = pool.get_context_delegators();
         tracker.ctx_pool(ctxd);
 
-        let batch_row_limit = {
-            if !on_resp.is_streaming() {
-                self.batch_row_limit
-            } else {
-                self.stream_batch_row_limit
+        let handler: Result<Box<RequestHandler>> = req_handler_builder(snap);
+
+        match handler {
+            Err(e) => {
+                on_resp.respond(err_resp(e, metrics));
             }
-        };
+            Ok(mut handler) => {
+                if let Err(e) = req_ctx.check_if_outdated() {
```

Comment:
I think this check is redundant since L18-126 happens immediately.

Reply:
Nice catch! We double checked this even before this PR!